lib/puppet/util/windows/process.rb

```diff
@@ -60,6 +64,30 @@ def get_current_process
   end
   module_function :get_current_process
 
+  def open_process(desired_access, inherit_handle, process_id, &block)
+    phandle = nil
+    begin
+      FFI::MemoryPointer.new(:handle, 1) do |phandle_ptr|
+        inherit = inherit_handle ? FFI::WIN32_TRUE : FFI::WIN32_FALSE
+        phandle_ptr = OpenProcess(desired_access, inherit, process_id)
+        if phandle_ptr == FFI::Pointer::NULL_HANDLE
+          raise Puppet::Util::Windows::Error.new(
+            "OpenProcess(#{desired_access.to_s(8)}, #{inherit}, #{process_id})")
+        end
+
+        yield phandle = phandle_ptr.read_handle
+      end
+
+      phandle
```

Comment:
I think we can drop this, and the `phandle =` assignment above.

Reply:
Oh, you're right! The block-passing was a remnant from an approach I was trying out earlier.

Reply:
Disregard my last comment, the block is still necessary.

Reply:
I've been unable to get tests to pass without this, I think phandle needs to be defined outside of the block in order for the CloseHandle ensure section to work. When I tried doing this with

`FFI::WIN32.CloseHandle(phandle_ptr) if phandle_ptr`

Ruby wasn't happy.